include/swift/AST/AutoDiff.h

```diff
@@ -208,6 +208,41 @@ struct AutoDiffDerivativeFunctionKind {
   }
 };
 
+/// Identifies an autodiff derivative function configuration:
+/// - Parameter indices.
+/// - Result indices.
+/// - Derivative generic signature (optional).
+// TODO(TF-893): Use `AutoDiffConfig` in `AutoDiffDerivativeFunctionIdentifier`
+// to avoid duplication.
+class AutoDiffConfig : public llvm::FoldingSetNode {
```

Comment:
* `AutoDiffConfig` does not need to be uniqued. It is only 3 words long and can simply be a POD.
* Fields do not need to be `const`.
* No getters are necessary.

```swift
struct AutoDiffConfig {
  IndexSubset *parameterIndices;
  IndexSubset *resultIndices;
  GenericSignature *derivativeGenericSignature;
};
```

Reply:
Chatted in person; POD makes sense because all three fields are already individually uniqued. `AutoDiffDerivativeFunctionIdentifier` is not necessarily a good example and may be removed later when VJPs are removed and only JVPs exist.

Done in https://github.com/apple/swift/pull/27487/commits/10700e714f16dfb35ff8930e55493976f0a829f2.